build.gradle

```diff
@@ -8,6 +8,156 @@ repositories {
+
+		if(process != null && waitfor==true)
+			process.waitFor()
+	}
+}
+
+task killAllTestProcesses {
+    def Platform = "${System.properties['os.name'].toLowerCase()}" 
+    
+    def ports = [6000, 8000]
+    ports.each { port ->
+     	if (Platform.contains('linux')){
+	        def cmd = "lsof -Fp -i :$port"
+	        def process = cmd.execute()
+	        process.in.eachLine { line ->
+	        	def command = "kill -9 "+line.substring(1)
+	        	def killProcess = command.execute()
+	            killProcess.waitFor()
+	        }
+	    }
+	    else if(Platform.contains('windows')) {
+	        def cmd = "netstat -a -o -n | findstr \"TCP\" | findstr \":$port\""
+	        def process = cmd.execute()
+	        process.in.eachLine { line ->
+	        	def command = "taskkill /F /PID  "+line.split('+ ')[4]
+	        	def killProcess = command.execute()
+	            killProcess.waitFor()
+	        }
+	    }
+
+    }
+}
+
+task checkLogFolder {
+	File logFolder = new File("./log/")
+	if( !logFolder.exists() ) {
+		logFolder.mkdirs()
+	}
+}
+
+task getNodeModules(type: ExecWait) {
+	command 'npm install'
+	ready 'Storage server modules initialised.'
+	directory './qabel-storage'
+	logname 'npm'
+	waitfor true
+}
+
+task checkStorage {
+	dependsOn 'checkLogFolder'
+	File storagedataFolder = new File("./qabel-storage/data/")
+	if( !storagedataFolder.exists() ) {
+		storagedataFolder.mkdirs()
+	}
+
+	File nodeModulesFolder = new File("./qabel-storage/node_modules/")
+	if( !nodeModulesFolder.exists() ) {
+		getNodeModules.execute()
+	}
+}
+
+task startStorageServerUnix(type: ExecWait) {
+	dependsOn 'checkStorage'
+	def cmd = "./node.sh"
+	def process = cmd.execute()
+	process.in.eachLine { line ->
+		command line+' app.js &'
+	}
+	ready 'Storage server running.'
+	directory 'qabel-storage/'
+	logname 'qabel-storage'
+}
+
+task startStorageServerWindows(type: ExecWait) {
+	dependsOn 'checkStorage'
+	command 'node app.js &'
+	ready 'Storage server running.'
+	directory 'qabel-storage'
+	logname 'qabel-storage'
+}
+
+task startDropServer(type: ExecWait) {
+	dependsOn 'checkLogFolder'
+	command 'python drop_server.py &'
+	ready 'Drop server running.'
+	directory 'qabel-drop'
+	logname 'qabel-drop'
+}
+
+task startServers {
+	def Platform = "${System.properties['os.name'].toLowerCase()}" 
+
+	startDropServer.execute()
+	if (Platform.contains('windows')){
+		startStorageServerWindows.execute()
+	}
+	else {
+		startStorageServerUnix.execute()
+	}
+}
+
+gradle.addListener new TestLifecycleListener()
+ 
+class TestLifecycleListener implements TaskExecutionListener {
+
+    @Override
+    void beforeExecute(Task task) {
+        if (task.name == 'test') {
+	        
+              task.project.rootProject.startServers.execute();
+        }
+    }
+ 
+    @Override
+    void afterExecute(Task task, TaskState taskState) {
+        if (task.name == 'test') {
+           task.project.rootProject.killAllTestProcesses.execute()
+        }
+    }
+}
+
+
+
+
+test {
```

Comment:
Can you tidy up this? It is empty and it works without it. Also remove the empty lines above.